### PR TITLE
added skip in ecdsa verify if eddsaph

### PR DESCRIPTION
**Pull request by PFnord**

### CODE REVIEW

File: oberon/drivers/oberon_ecdsa.c [hunk 1]
```diff
@@ -477,7 +477,7 @@ psa_status_t oberon_ecdsa_verify_message(
     psa_key_type_t type = psa_get_key_type(attributes);
 
     switch (type) {
-#if defined(PSA_NEED_OBERON_PURE_EDDSA_TWISTED_EDWARDS_255) || defined(PSA_NEED_OBERON_PURE_EDDSA_TWISTED_EDWARDS_448)
+#if (defined(PSA_NEED_OBERON_PURE_EDDSA_TWISTED_EDWARDS_255) || defined(PSA_NEED_OBERON_PURE_EDDSA_TWISTED_EDWARDS_448)) && !defined(PSA_NEED_OBERON_ED25519PH)
```
Comment (magnev) [inline oberon/drivers/oberon_ecdsa.c]: I don't think we can have this extra ifdef here.
We must still be able to run EdDSA pure if PSA_NEED_OBERON_ED25519PH is enabled.

File: oberon/drivers/oberon_ecdsa.c [hunk 2]
```diff
@@ -477,7 +477,7 @@ psa_status_t oberon_ecdsa_verify_message(
...
     case PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_TWISTED_EDWARDS):
     case PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_FAMILY_TWISTED_EDWARDS):
         // EDDSA is only available in verify_message
```
Comment (magnev) [inline oberon/drivers/oberon_ecdsa.c]: If you add a check here to make sure the algorithm is EDDSA PURE it should work. Something like:
`if (alg != PSA_ALG_PURE_EDDSA) return PSA_ERROR_NOT_SUPPORTED;`

Comment (PFnord) [reply oberon/drivers/oberon_ecdsa.c]: Do you mean `if (alg != PSA_ALG_ED25519PH)`? We do want to support PURE EDDSA still no?

File: oberon/drivers/oberon_ecdsa.c (hunk 1 above)
Comment (PFnord) [reply oberon/drivers/oberon_ecdsa.c]: Yes true, the config being enabled doesn't necessarily mean it's ED25519ph

File: oberon/drivers/oberon_ecdsa.c [hunk 3]
```diff
@@ -304,7 +306,9 @@ psa_status_t oberon_ecdsa_sign_message(
 #endif
 #ifdef PSA_NEED_OBERON_PURE_EDDSA_TWISTED_EDWARDS_448
         case 448:
-            if (key_length != ocrypto_ed448_SECRET_KEY_BYTES) return PSA_ERROR_INVALID_ARGUMENT;
+            if (alg == PSA_ALG_ED448PH) return PSA_ERROR_NOT_SUPPORTED;
+
+			if (key_length != ocrypto_ed448_SECRET_KEY_BYTES) return PSA_ERROR_INVALID_ARGUMENT;
```
Comment (frkv) [inline oberon/drivers/oberon_ecdsa.c]: Something off with the intentation here

File: oberon/drivers/oberon_ecdsa.c [hunk 4]
```diff
@@ -484,6 +488,8 @@ psa_status_t oberon_ecdsa_verify_message(
         switch (psa_get_key_bits(attributes)) {
 #ifdef PSA_NEED_OBERON_PURE_EDDSA_TWISTED_EDWARDS_255
         case 255:
+            if (alg == PSA_ALG_ED25519PH) return PSA_ERROR_NOT_SUPPORTED;
+
```
Comment (frkv) [inline oberon/drivers/oberon_ecdsa.c]: The other checks don't have this newline, so please skip it

File: oberon/drivers/oberon_ecdsa.c [hunk 5]
```diff
@@ -495,6 +501,8 @@ psa_status_t oberon_ecdsa_verify_message(
 #endif /* PSA_NEED_OBERON_PURE_EDDSA_TWISTED_EDWARDS_255 */
 #ifdef PSA_NEED_OBERON_PURE_EDDSA_TWISTED_EDWARDS_448
         case 448:
+            if (alg == PSA_ALG_ED448PH) return PSA_ERROR_NOT_SUPPORTED;
+
```
Comment (frkv) [inline oberon/drivers/oberon_ecdsa.c]: The other check don't have this newline, please skip it

File: oberon/drivers/oberon_ecdsa.c [hunk 6]
```diff
@@ -295,6 +295,8 @@ psa_status_t oberon_ecdsa_sign_message(
         switch (psa_get_key_bits(attributes)) {
 #ifdef PSA_NEED_OBERON_PURE_EDDSA_TWISTED_EDWARDS_255
         case 255:
+            if (alg == PSA_ALG_ED25519PH) return PSA_ERROR_NOT_SUPPORTED;
+
```
Comment (frkv) [inline oberon/drivers/oberon_ecdsa.c]: The other checks don't have this newline, so please skip it

**Review by frkv — CHANGES_REQUESTED** [review]

**Review by magnev — APPROVED** [review]

**Review by frkv — COMMENTED** [review]
LGTM

**Review by frkv — APPROVED** [review]

**Pull request merged**